### DISCUSSION TITLE
Can ponder work with something like info.uniswap.org?

### DISCUSSION THREAD

**Discussion by uniyj** [ponder-sh/ponder#157] [General]
Just wondering if ponder can be used by something like info.uniswap.org, where ponder will need to index and serve large volume of historical data starting from genesis of Uniswap Pool contracts?

(Might be a dumb question, sorry about that)

yj

**Comment by typedarray** [comment]
That particular app likely requires a feature that Ponder doesn't support yet (dynamic addition of event log sources). But yes, the intent is for Ponder to support apps like a Uniswap frontend.

**Comment by Artelenna** [comment]
very beautiful)

**Comment by valensov** [comment]
Interesting thoughts

**Comment by kalinkavita** [comment]
Very interesting

**Comment by AminatFialka** [comment]
Very helpful, thank you!)

**Comment by Vactions6** [comment]
good idea

**Comment by Starla61422457** [comment]
Interesting!

**Comment by Kasian8702** [comment]
I have a same question

**Comment by LunaLi777** [comment]
I'm listening to wise people)

**Comment by Djagernaut11** [comment]
nice

**Comment by adamgall** [comment]
@typedarray is this possible in Ponder yet? I don't see where in the docs it's possible to dynamically create event listeners.

edit: just saw [this open issue](https://github.com/ponder-sh/ponder/issues/1320). seems like it's still in progress then?

**Reply by typedarray** [reply]
Yes, it is - this is solved with our factory feature (https://ponder.sh/docs/contracts-and-networks#factory-contracts).

**Reply by adamgall** [reply]
Ah yes ok... I think my specific use case is a bit different than this, but I appreciate your response.

**Reply by typedarray** [reply]
What's your specific use case? Might be able to suggest a workaround.

**Reply by adamgall** [reply]
ok! bear with me, this is a doozie:

- start by using the factory feature to watch the [SafeProxyFactory](https://github.com/safe-global/safe-smart-account/blob/v1.4.1/contracts/proxies/SafeProxyFactory.sol) contract, to get all instances of Safes that are created
- for each Safe, I listen to the [`EnabledModule`](https://github.com/safe-global/safe-smart-account/blob/v1.4.1/contracts/base/ModuleManager.sol#L54) event (this is supported in ponder through that factory feature)
- then, I need to do some of my own introspection on each of those module addresses which come from the `EnabledModule` event. If the given module address implements a specific interface, I want to set up some more ponder event listeners on it.
  - this is where I'm stuck. For the sake of this discussion, I don't know who created those modules, or how they were deployed. I only care that they support some interface (I can write the code to check for this interface support).
